internal/output/aws_credentials_file.go

```diff
@@ -179,15 +183,16 @@ func NewAWSCredentialsFile(legacyVars bool, expiryVars bool, expiry string) *AWS
 
 // Output Satisfies the Outputter interface and appends AWS credentials to
 // credentials file.
-func (e *AWSCredentialsFile) Output(c *config.Config, oc *oaws.Credential, ac *sts.Credentials) error {
+func (a *AWSCredentialsFile) Output(c *config.Config, oc oaws.Credential) error {
+	cfc := oc.(*oaws.CredsFileCredential)
```

Comment:
I really don't like that we have to cast things into concrete type from an interface{}. That has already introduce a lot of issue in the tf provider. Even in the v3 of tfsdk framework, they removed that. I would much rather you defined whatever field you need in the interface and exposing them that way